llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -632,12 +632,15 @@ class DXILResourceBindingInfo {
     RegisterSpace(uint32_t Space) : Space(Space) {
       FreeRanges.emplace_back(0, UINT32_MAX);
     }
+    // Size == -1 means unbounded array
+    bool findAvailableBinding(int32_t Size, uint32_t *RegSlot);
```

Comment:
Better to use `std::optional<uint32_t>` here instead, even if it's just for clarity.

It looks like dxc falls over if we have bindings with sizes greater than a signed int anyway, but at least in principle I think they should be allowed.